pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/TabularParticipantExporter.java

```diff
@@ -0,0 +1,128 @@
+    protected List<Map<String, String>> participantValueMaps;
+
+    public abstract void export(Response response) throws IOException;
+
+    protected abstract String sanitizeValue(String value);
+
+    protected TabularParticipantExporter(List<ModuleExportConfig> moduleConfigs,
+                                         List<Map<String, String>> participantValueMaps, String fileFormat) {
+        this.moduleConfigs = moduleConfigs;
+        this.participantValueMaps = participantValueMaps;
+        this.fileFormat = fileFormat;
+    }
+
+    public static TabularParticipantExporter getExporter(List<ModuleExportConfig> moduleConfigs,
+                                                         List<Map<String, String>> participantValueMaps, String fileFormat) {
+        if (TSV_FORMAT.equals(fileFormat)) {
+            return new TsvParticipantExporter(moduleConfigs, participantValueMaps, fileFormat);
+        } else if (XLSX_FORMAT.equals(fileFormat)) {
+            return new ExcelParticipantExporter(moduleConfigs, participantValueMaps, fileFormat);
+        }
+        throw new RuntimeException("Unrecognized file format");
+    }
+
+    protected static String getExportFilename(String suffix) {
+        LocalDate date = LocalDate.now();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(FILE_DATE_FORMAT);
+        String exportFileName = String.format("Participant-%s.%s", date.format(formatter), suffix);
+        return exportFileName;
+    }
+
+    public static List<String> getAllColumnNames(FilterExportConfig filterConfig, int formRepeatNum, int questionRepeatNum) {
+        if (filterConfig.isSplitOptionsIntoColumns()) {
+            return filterConfig.getOptions().stream().map(opt -> {
+                return getColumnName(
+                        filterConfig,
+                        formRepeatNum,
+                        questionRepeatNum,
+                        opt);
+            }).collect(Collectors.toList());
+        }
+        return Collections.singletonList(getColumnName(
+                filterConfig,
+                formRepeatNum,
+                questionRepeatNum,
+                null
+        ));
+    }
+
+    public static String getColumnName(FilterExportConfig filterConfig,
+                                       int activityRepeatNum,
+                                       int questionRepeatNum,
+                                       Map<String, Object> option) {
+        String activityName = filterConfig.getParent().getName();
+        String questionStableId = filterConfig.getColumn().getName();
+        String activityExportName = activityRepeatNum > 1 ?
+                activityName + COLUMN_REPEAT_DELIMITER + activityRepeatNum : activityName;
+        String columnExportName = questionRepeatNum > 1 ?
+                questionStableId + COLUMN_REPEAT_DELIMITER + questionRepeatNum : questionStableId;
+        if (option != null && filterConfig.isSplitOptionsIntoColumns()) {
```

Comment:
I personally prefer to extract conditions like this as a method with meaningful name for quick understanding, in this case something like: `hasOption` if I got the idea correctly.